### ISSUE TITLE
[Core Dump] HHVM fails when an array splat is used as an argument to a reified function

### ISSUE THREAD

**Issue by lexidor** [facebook/hhvm#8549]
### HHVM Version

```
HipHop VM 4.17.1 (rel)
Compiler: 1565219722_353117164
Repo schema: 852e74da865cb32059208f20e096206fc2a11483
```
I'll install dbg after this to give a more detailed core dump.

### Operating System and Version
`Ubuntu 18.04`

### Standalone code, or other way to reproduce the problem
```HACK
<<__EntryPoint>>
function dump_core_async(): void {
  reified<string>(...['title']);
}

function reified<reify T>(
  string $_,
): void { }
```

### Actual result
```
Core dumped: Segmentation fault
Stack trace in /tmp/stacktrace.15014.log
Segmentation fault (core dumped)
```

### Expected result
nothing, end of script

**Comment by lexidor** [comment]
```
HipHop VM 4.17.1 (rel)
Compiler: 1565219722_353117164
Repo schema: 852e74da865cb32059208f20e096206fc2a11483
```

```
Host: lexidor
ProcessID: 15014
ThreadID: 140256958426240
ThreadPID: 15014
Name: /usr/bin/hhvm
CmdLine: hhvm start.hack 
Type: Segmentation fault
Runtime: hhvm
Version: 1565219722_353117164
DebuggerCount: 0

Arguments: start.hack
ThreadType: CLI

# 0  0000563529326fc6
# 1  00005635295db244
# 2  00007f901c062890
# 3  00005635298fe06f
# 4  000056352991460a
# 5  000056352991498a
# 6  000056352dc002a8
# 7  000056352dc00351
# 8  000056352ae694f4
# 9  000056352ae691e3
# 10 000056352ae68847
# 11 000056352af34a1a
# 12 000056352af3477c
# 13 000056352af344c0
# 14 000056352af330de
# 15 0000563529676f15
# 16 0000563529680cff
# 17 0000563529682fbe
# 18 00005635292b092a
# 19 00007f901603fb97
# 20 00005635292ae97a

PHP Stacktrace:
```

**Comment by lexidor** [comment]
From hhvm-dbg
```
Host: lexidor
ProcessID: 4320
ThreadID: 139834835004544
ThreadPID: 4320
Name: /usr/bin/hhvm
CmdLine: hhvm start.hack 
Type: Segmentation fault
Runtime: hhvm
Version: 1565219722_353117164
DebuggerCount: 0

Arguments: start.hack
ThreadType: CLI

# 0  0000563aac107fc6
# 1  0000563aac3bc244
# 2  00007f2dd381f890
# 3  0000563aac6df06f
# 4  0000563aac6f560a
# 5  0000563aac6f598a
# 6  0000563ab02002a8
# 7  0000563ab0200351
# 8  0000563aadc4a4f4
# 9  0000563aadc4a1e3
# 10 0000563aadc49847
# 11 0000563aadd15a1a
# 12 0000563aadd1577c
# 13 0000563aadd154c0
# 14 0000563aadd140de
# 15 0000563aac457f15
# 16 0000563aac461cff
# 17 0000563aac463fbe
# 18 0000563aac09192a
# 19 00007f2dcd7fcb97
# 20 0000563aac08f97a

PHP Stacktrace:
```

**Comment by fredemmott** [comment]
FB references D17352274 D17354372 (will also show up in commit messages)

**Comment by fredemmott** [comment]
landed, still repros

**Comment by fredemmott** [comment]
non-splat variadics appear fixed, splat still an issue

**Comment by fredemmott** [comment]
FB T54142576

**Comment by fredemmott** [comment]
No longer repros, fixed in nightly-2019.09.26. ca0701697a looks the most likely
